esmvalcore/preprocessor/_multimodel.py

```diff
@@ -248,115 +97,206 @@ def _unify_time_coordinates(cubes):
+    cube : :obj:`iris.cube.Cube`
+        Input cube.
+    statistic : str
+        Name of the statistic to calculate. Must be available via
+        :mod:`iris.analysis`.
+    dim : str
+        Collapse cube along this coordinate.
 
     Returns
     -------
-    dict
-        dictionary of statistics cubes with statistics' names as keys.
-
-    Raises
-    ------
-    ValueError
-        If span is neither overlap nor full.
+    :obj:`iris.cube.Cube`
+        Collapsed cube.
     """
-    if len(cubes) < 2:
-        logger.info('Found only 1 cube; no statistics computed for %r',
-                    list(cubes)[0])
-        return {statistic: cubes[0] for statistic in statistics}
+    statistic = statistic.lower()
+    kwargs = {}
 
-    logger.debug('Multicube statistics: computing: %s', statistics)
+    # special cases
+    if statistic == 'std':
+        logger.warning(
+            "Multicube statistics is aligning its behaviour with iris.analysis"
+            ". Please consider replacing 'std' with 'std_dev' in your code.")
+        statistic = 'std_dev'
 
-    # Reset time coordinates and make cubes share the same calendar
-    _unify_time_coordinates(cubes)
+    elif re.match(r"^(p\d{1,2})(\.\d*)?$", statistic):
+        # percentiles between p0 and p99.99999...
+        percentile = float(statistic[1:])
+        kwargs['percent'] = percentile
+        statistic = 'percentile'
 
-    # Check whether input is valid
-    if span == 'overlap':
-        # check if we have any time overlap
-        times = [cube.coord('time').points for cube in cubes]
-        overlap = reduce(np.intersect1d, times)
-        if len(overlap) <= 1:
-            logger.info("Time overlap between cubes is none or a single point."
-                        "check datasets: will not compute statistics.")
-            return cubes
-        logger.debug("Using common time overlap between "
-                     "datasets to compute statistics.")
-    elif span == 'full':
-        logger.debug("Using full time spans to compute statistics.")
-    else:
+    try:
+        operator = STATISTIC_MAPPING[statistic]
+    except KeyError as err:
         raise ValueError(
-            "Unexpected value for span {}, choose from 'overlap', 'full'".
-            format(span))
+            f'Statistic `{statistic}` not supported by multicube statistics. '
+            f'Must be one of {tuple(STATISTIC_MAPPING.keys())}.') from err
+
+    logger.debug('Multicube statistics: computing: %s', statistic)
+
+    # This will always return a masked array
+    return cube.collapsed(dim, operator, **kwargs)
 
-    # Compute statistics
+
+def _multicube_statistics(cubes, statistics, span):
+    """Compute statistics over multiple cubes.
+
+    Can be used e.g. for ensemble or multi-model statistics.
+
+    Cubes are merged and subsequently collapsed along a new auxiliary
+    coordinate. Inconsistent attributes will be removed.
+    """
+    aligned_cubes = _align(cubes, span=span)
```

Comment:
Maybe copy the original cubes so we don't modify them inplace?

Reply:
Good point! Do you think this will have performance implications for large non-lazy datasets?